modules/siddhi-core/src/main/java/io/siddhi/core/table/record/AbstractQueryableRecordTable.java

```diff
@@ -119,16 +137,45 @@ public void initCache(TableDefinition tableDefinition, SiddhiAppContext siddhiAp
                     cacheTableDefinition.annotation(annotation);
                 }
             }
-            cachedTable.initTable(cacheTableDefinition, storeEventPool,
-                    storeEventCloner, configReader, siddhiAppContext, recordTableHandler);
 
+            String cachePolicy = cacheTableAnnotation.getElement(ANNOTATION_CACHE_POLICY);
+
+            if (cachePolicy == null || cachePolicy.equalsIgnoreCase("FIFO")) {
+                cachePolicy = "FIFO";
+                cacheTable = new CacheTableFIFO();
+            } else if (cachePolicy.equalsIgnoreCase("LRU")) {
+                cacheTable = new CacheTableLRU();
+            } else if (cachePolicy.equalsIgnoreCase("LFU")) {
+                cacheTable = new CacheTableLFU();
+            }
```

Comment:
Need to through an error when a wrong value other than the above 3 are given. 
The error should say found cachePolicy `foo` by only supports FIFO, ....